src/Simulation/Native/src/external/fusion.hpp

```diff
@@ -20,6 +21,11 @@ class Item{
 	Item(Matrix mat, IndexVector idx) : mat_(mat), idx_(idx) {}
```

Comment:
Thank you for pointing this out. However, this change takes us away a similar design pattern used throughtout the repository. This change would require a significant overhaul in many aspects of the codebase. We could look into making use of std:move, however it would be appropriate to get back to this as part of a another PR linking it to an appropriate issue.

Reply:
You can avoid the extra copy by a one line fix inside the constructor which doesn't affect any of the callers and makes the code more efficient. Having multiple bugs of this type doesn't make any one of them less buggy. However, I do agree that this is a pre-existing code and _not_ fixing it as part of this PR is a valid choice.

Reply:
I think Aniket's point here is that there are other constructors throughout the simulator that should also be updated to follow this. Rather than have one constructor use one pattern while the others use the old one, we should do a single change that updates all the constructors to use `std::move` that is not tied to this PR. That makes sure that the pattern is fixed everywhere at once and makes it easier to review and discuss that change without having get mixed in with this PR that is meant to introduce the new functionality.

Reply:
As I said above, not fixing pre-existing code with this PR is a valid choice. I still believe that this pattern is sub-optimal and shouldn't be used to justify new code.